test/test_string_utils.cpp

```diff
@@ -0,0 +1,47 @@
+/* Copyright (C) 2021, Nikolai Wuttke. All rights reserved.
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <base/warnings.hpp>
+#include <common/string_utils.hpp>
+
+RIGEL_DISABLE_WARNINGS
+#include <catch.hpp>
+RIGEL_RESTORE_WARNINGS
+
+TEST_CASE("String split") {
+  SECTION("Empty string") {
+    std::vector<std::string> v;
+    rigel::strings::split(v, "", ' ');
+    CHECK(v.size() == 1);
+  }
+  SECTION("String with no delimiter") {
+    std::vector<std::string> v;
+    rigel::strings::split(v, "hello", ' ');
+    CHECK(v.size() == 1);
+  }
+  SECTION("String with delimiter") {
+    std::vector<std::string> v;
+    rigel::strings::split(v, "hello, world", ',');
+    CHECK(v.size() == 2);
+    CHECK(v.front() == "hello");
+    CHECK(v.back() == " world");
+  }
+  SECTION("String with other white space delimiters") {
+    std::vector<std::string> v;
+    rigel::strings::split(v, "hello\nworld", '\n');
+    CHECK(v.size() == 2);
+  }
+}
```

Comment:
Another edge case that could be worth testing is a string that ends with a delimiter, like `"hello\n".